src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/comments/ProblematicYamlTest.kt

```diff
@@ -0,0 +1,168 @@
+        val yamlString3 = "" +
+            "key: value\n" +
+            "\n" +
+            "key: value\n"
+
+        val expectedEventIdList = listOf(
+            Event.ID.StreamStart,
+            Event.ID.DocumentStart,
+            Event.ID.MappingStart,
+            Event.ID.Scalar,
+            Event.ID.Scalar,
+            Event.ID.Comment,
+            Event.ID.Scalar,
+            Event.ID.Scalar,
+            Event.ID.MappingEnd,
+            Event.ID.DocumentEnd,
+            Event.ID.StreamEnd
+        )
+        val expectedCommentTypeList = listOf(CommentType.BLANK_LINE)
+        val parser = ParserImpl(LOAD_OPTIONS, StreamReader(LOAD_OPTIONS, yamlString3))
+        assertEventListEquals(expectedEventIdList, expectedCommentTypeList, parser)
+    }
+
+    test("parse problematic yaml 4") {
+        val yamlString4 = "" +
+            "---\n" +
+            "in the block context:\n" +
+            "    indentation should be kept: { \n" +
+            "    but in the flow context: [\n" +
+            "it may be violated]\n" +
+            "}\n" +
+            "---\n" +
+            "the parser does not require scalars\n" +
+            "to be indented with at least one space\n" +
+            "...\n" +
+            "---\n" +
+            "\"the parser does not require scalars\n" +
+            "to be indented with at least one space\"\n" +
+            "---\n" +
+            "foo:\n" +
+            "    bar: 'quoted scalars\n" +
+            "may not adhere indentation'\n"
+
+        val expectedEventIdList = listOf(
+            Event.ID.StreamStart,
+            Event.ID.DocumentStart,
+            Event.ID.MappingStart,
+            Event.ID.Scalar,
+            Event.ID.MappingStart,
+            Event.ID.Scalar,
+            Event.ID.MappingStart,
+            Event.ID.Scalar,
+            Event.ID.SequenceStart,
+            Event.ID.Scalar,
+            Event.ID.SequenceEnd,
+            Event.ID.MappingEnd,
+            Event.ID.MappingEnd,
+            Event.ID.MappingEnd,
+            Event.ID.DocumentEnd,
+            Event.ID.DocumentStart,
+            Event.ID.Scalar,
+            Event.ID.DocumentEnd,
+            Event.ID.DocumentStart,
+            Event.ID.Scalar,
+            Event.ID.DocumentEnd,
+            Event.ID.DocumentStart,
+            Event.ID.MappingStart,
+            Event.ID.Scalar,
+            Event.ID.MappingStart,
+            Event.ID.Scalar,
+            Event.ID.Scalar,
+            Event.ID.MappingEnd,
+            Event.ID.MappingEnd,
+            Event.ID.DocumentEnd,
+            Event.ID.StreamEnd
+        )
+        val settings = LoadSettings.Companion.builder().build()
+        val parser = ParserImpl(settings, StreamReader(settings, yamlString4))
+        assertEventListEquals(expectedEventIdList, emptyList(), parser)
+    }
+})
+
+private fun assertEventListEquals(
+    expectedEventIdList: List<Event.ID>,
+    expectedCommentTypeList: List<CommentType>,
+    parser: Parser
```

Comment:
I would make it a receiver for that function. In this case, the call side would look like this
```kotlin
ParserImpl(settings, StreamReader(settings, yamlString4))
  .assertEventListEquals(expectedEventIdList, emptyList())
```
and we don't need an extra variable in between